src/cryptonote_core/tx_pool.cpp

```diff
@@ -780,7 +819,7 @@ namespace cryptonote
     const uint64_t now = time(NULL);
     txs.reserve(m_blockchain.get_txpool_tx_count());
     m_blockchain.for_all_txpool_txes([this, now, &txs](const crypto::hash &txid, const txpool_tx_meta_t &meta, const cryptonote::blobdata *){
-      if(!meta.do_not_relay && now - meta.last_relayed_time > get_relay_delay(now, meta.receive_time))
+      if(!meta.do_not_relay && (!meta.relayed || now - meta.last_relayed_time > get_relay_delay(now, meta.receive_time)))
```

Comment:
Adding !meta.relayed will always cancel out the 2nd condition of 
`(!meta.relayed || now - meta.last_relayed_time > get_relay_delay(now, meta.receive_time))` 

which is to try and re-relay transactions after some time (even those that have previously been sent succesfully). After a set_relayed() all tx's in the pool will have `relayed = true`. At the point `relay_transactions` is called there's no check if the relay failed so no relay after the first initial relay.

Why do we want to only relay once?

Reply:
I see, the `pool.relay_txpool_transactions()` in `process_blink_signatures`
Perhaps we want a `relay_tx(hash)` to avoid flooding the network on each received blink?

Reply:
> Adding !meta.relayed will always cancel out the 2nd condition of ...

I'm not following this.

The logic here (ignoring the `do_not_relay` bit) is:

if ([never relayed] || [relayed a long time ago])

Reply:
O yeah nvm misread it as an &&.

Reply:
> Perhaps we want a relay_tx(hash) to avoid flooding the network on each received blink?

That flood happens already on every received regular tx (it sets `should_be_relayed` to true when the tx gets added to the mempool when adding a regular tx without `do_not_relay`, which triggers an immediate relay).